[Chew Cheng Yap] iP

CodyChew opened the pull request.

Commit:
Add Gradle support

CodyChew committed:
Completed level 1:
-skeletal version of Dukenizer chatbot
-Included echoing of commands and exits when user types "bye"

CodyChew committed:
Completed Level 2:
-Add items into a list
-display list when called

CodyChew committed:
Completed Level 3:
-added task class
-show active/done tasks

CodyChew committed:
Completed Level 4:
-Included Todo, Event and Deadline Class
-Each task are labeled with the starting alphabet

CodyChew committed:
Automated Text UI Testing:
Included runtest.bat

CodyChew committed:
Update README.md

CodyChew committed:
Major code refactoring

CodyChew committed:
Fix runtest.bat
-now checks for tick/cross correctly

CodyChew committed:
Merge branch 'master' of https://github.com/CodyChew/ip

CodyChew committed:
Level 5 complete: Handle Errors
-Additional method to throw DukeException while checking instruction line
-todo, deadline, event cannot be empty
-done must have more than 1 argument separated with space, is number specified cannot be more than Task list
-Many more exceptions have yet to be implemented

CodyChew committed:
Handle more exceptions:
-All exceptions are handled when parsing the instruction
-leading and trailing white spaces are stripped for every line

following commands:
list:
-checks for no other inputs, (only "list")

done:
-checks for more than 1 input separated by space
-checks for second input being a valid number and in the following format 'done <item number>'

todo:
-checks for no other inputs, (only "todo")

deadline:
-checks for 'deadline <description> /by <time>' format

event:
-checks for 'event <description> /at <time>' format

CodyChew committed:
Level 6 Completed:
-added delete task from list
-delete exceptions uses same checks as done exceptions

CodyChew committed:
Code refactoring:
-shifted execution of instruction out of main method into read instruction method
-separated delete and done logic for more customisations in the future

CodyChew committed:
Add saving tasks to hard disk

Let's write to a relative file path to a text file Dukenizer.txt. Every
time the list updates, the file is written again.

This is to easily get the full list string everytime the list is updated
without having to process line by line when writing to the file.

CodyChew committed:
Update deadline and event command to read Date and Time

Represent DataTime as an object rather than a string for manipulation.

Let's store a LocalDateTime obejct for the Deadline and Event task.

Able to format its output easily.

Reference to validation for correct dates and time format
https://stackoverflow.com/questions/2149680/regex-date-format-validation-on-java
https://www.geeksforgeeks.org/how-to-validate-time-in-24-hour-format-using-regular-expression/

CodyChew committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Duke.java

CodyChew committed:
Refactor code to extract out Ui, Storage, Parser, TaskList, Command classes

Classes have closely related code can be extracted into a single class.
It reduces the complexity of viewing things.

Ui class: deals with interactions with the user.

Storage class: deals with loading tasks from the file and saving tasks
in the file.

Parser class: deals with making sense of the user command.

TaskList class: contains the task list. It has operations to manipulate
tasks in the list and get data from it.

Command class: abstract Command class to allow us to use polymorphism
when dealing with different commands

CodyChew committed:
Repackage classes

Organize class types into packages for easier management

packages:
-Duke
-Duke.command
-Duke.exception
-Duke.parser
-Duke.storage
-Duke.task
-Duke.ui

CodyChew committed:
Add JUnit tests for TaskList class

CodyChew committed:
Add JavaDoc to Duke, Command, Parser, Storage, Ui, Task, TaskList,
DukeException classes

CodyChew committed:
Refactor code to comply with coding standard

Refer to Java coding standard
https://se-education.org/guides/conventions/java/index.html

CodyChew committed:
Add Find command

Give users a way to find a task from its description with a search term.

Let's add a FindCommand class and updated Parser to accomodate find
command keyword.

CodyChew committed:
Merge branch 'branch-A-JavaDoc'

CodyChew committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/command/ListCommand.java
#	src/main/java/duke/storage/Storage.java

CodyChew committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/command/FindCommand.java

CodyChew committed:
Merge branch 'add-gradle-support' of https://github.com/CodyChew/ip into branch-A-Gradle

CodyChew committed:
Update main class name in build.gradle

CodyChew committed:
Add UTF-8 text in Gradle

Task done attribute makes use of UTF-8 encoding for task description.

Let's configure Gradle compiler encoding to 'UTF-8'.

Refer to article on setting UTF-8 properly in gradle:
https://www.thetopsites.net/article/58616121.shtml
Refer to this discussion on setting up UTF-8 to gradle:
https://github.com/nus-cs2103-AY2021S1/forum/issues/95

faqihisaninja commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -93,6 +93,13 @@ else if (instruction.length == 2) {
+				throw new DukeException("Please specify a description to search!");
+			} else {
+				return new FindCommand(instruction[1]);
+			}
```
Some of your logical blocks have a blank line before and after, some only before and some only after. Maybe you can make this consistent?

faqihisaninja commented on src/main/java/Duke/storage/Storage.java:
```diff
@@ -45,12 +61,21 @@ public String load() throws DukeException {
 
+	/**
+	 * Get file path of this Storage object when it is created.
+	 * @return Saved file path
```
Perhaps an empty line between description and return section?

faqihisaninja commented on src/main/java/Duke/task/TaskList.java:
```diff
@@ -43,19 +57,40 @@ public TaskList(String load) throws DukeException {
 
 
+	/**
+	 * String representation of the TaskList.
```
Perhaps add a verb at the start of this javadoc comment for this method?

faqihisaninja commented on src/main/java/Duke/task/TaskList.java:
```diff
@@ -82,15 +117,31 @@ public String toString() {
 	}
 
+	/**
+	 * Return size of the task list.
```
Maybe use the "Returns" instead of "Return"? The imperative form is also apparent in many other javadoc comments of other methods. Perhaps consider changing them to be conform to the java coding standards?

faqihisaninja commented on src/main/java/duke/task/Event.java:
```diff
@@ -25,6 +25,7 @@ public Event(String done, String description, String at) {
     public String toString() {
-        return "[E]" + super.toString() + " (at: " + at.format(DateTimeFormatter.ofPattern("MMM d yyyy HH:mm")) + ")";
+        return "[E]" + super.toString() + " (at: " +
+                at.format(DateTimeFormatter.ofPattern("MMM d yyyy HH:mm")) + ")";
```
I like that you placed line breaks in many parts of your code including this one. Improves readability. 👍

CodyChew committed:
Merge branch 'branch-A-Gradle'

CodyChew committed:
Add checkstyle support

Let's add configuration files for checkstyle and use checkstyle
with gradle. Changed all files to comply with Java coding standard.

CodyChew committed:
Merge branch 'branch-A-CheckStyle'

faqihisaninja commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -1,110 +1,113 @@
+            } else if (instruction.length == 2) { //done with exactly 2 inputs
+                return new DeleteCommand(instruction[1]);
+            }
+        } else if (instruction[0].equals("find")) {
```
Maybe consider using a switch statement for this long list of conditionals?

faqihisaninja reviewed (commented):
LGTM. Just some nits. The checkstyle was a nice addition. Now your code looks very neat! 💯

youyingfeng commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,113 @@
+        }
+
+        throw new DukeException("☹ OOPS!!! I'm sorry, but I don't know what that means :-(");
+    }
```
Perhaps try to shove some code into separate methods? This method seems way too long (although many others do the same).

youyingfeng commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,124 @@
+                e.printStackTrace();
+            }
+        }
+        throw new DukeException("File does not exist");
```
You might want to put this in a separate else clause, to show that this only happens if the file does not exist.

youyingfeng reviewed (commented):
Overall pretty good!

CodyChew committed:
Add GUI to project

Use javaFX to implement GUI for task manager. Refactor UI to return
String value rather than standard output to implement GUI.

CodyChew committed:
Merge branch 'branch-Level-10'

CodyChew committed:
Update application name in gradle

CodyChew committed:
Add Java Assertions

Document important assumptions that should hold at various points
in the code.

CodyChew committed:
Update Greeting message in GUI

CodyChew committed:
Add JUnit test for Duke class

Main test for correct string output based on user input.

CodyChew committed:
Refactor code to improve code quality

CodyChew committed:
Merge pull request #1 from CodyChew/branch-A-Assertions

Add Java Assertions

CodyChew committed:
Merge branch 'master' of https://github.com/CodyChew/ip

CodyChew committed:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/duke/task/TaskList.java

CodyChew committed:
Merge pull request #2 from CodyChew/branch-A-CodeQuality

Refactor code to improve code quality

CodyChew committed:
Merge branch 'master' of https://github.com/CodyChew/ip

CodyChew committed:
Update comments

CodyChew committed:
Refactor to use Java Streams

CodyChew committed:
Merge pull request #3 from CodyChew/branch-A-Streams

Refactor to use Java Streams

CodyChew committed:
Merge branch 'master' of https://github.com/CodyChew/ip

CodyChew committed:
Add Continuous Integration

CodyChew committed:
Remove I/O tests in CI tool

JUnit tests are used in this project.

CodyChew committed:
Add D-Notes extension

Able to add notes and delete notes in a separate note list.

CodyChew committed:
Merge branch 'branch-D-Notes'

CodyChew committed:
Minor bug fix to NoteList

CodyChew committed:
Add help command

help command enables users to view all available commands for
improved user experience.

CodyChew committed:
Change DialogBox styling

Let's differentiate user's and Dukenizer's dialog box. User's box is
now blue while Dukenizer's box is yellow.

CodyChew committed:
Add background colour to GUI.

CodyChew committed:
Add screenshot of product

CodyChew committed:
Update docs\README.md

CodyChew committed:
Update image position in docs\README.md

CodyChew committed:
Set theme jekyll-theme-merlot

CodyChew committed:
Set theme jekyll-theme-time-machine

CodyChew committed:
Set theme jekyll-theme-time-machine

CodyChew committed:
Update README.md

CodyChew committed:
Set theme jekyll-theme-merlot

CodyChew committed:
Merge branch 'master' of https://github.com/CodyChew/ip

CodyChew committed:
Update GUI

Remove logo in greeting message as there is inconsistent formatting.
Window size has increased and dialog box now has a consistent size
for better viewing.

CodyChew committed:
Add shut down feature to GUI

Program now shuts down on exit command.

Let's return the exit message to the user and then delay for a while
before closing the window.

Refer to this documentation on the preferred way to terminate a JavaFX
Application:
https://docs.oracle.com/javase/8/javafx/api/javafx/application/Application.html

CodyChew committed:
Update pictures for docs/README.md

CodyChew committed:
Update docs/README.md

CodyChew committed:
Update doc/README with jar executable link

CodyChew committed:
Update README.md
